File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, a: List[int]) -> int:
        ans = zeros = lo = 0
        for hi, x in enumerate(a):
            zeros += 1 - x
            while zeros > 1:
                zeros -= 1 - a[lo]
                lo += 1
            ans = max(ans, hi - lo)
        return ans
            
            
        

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] a) {
        int i = 0, zero = 0, ans = 0;
        for (int j = 0; j < a.length; j++) {
            zero += 1 - a[j];
            while (zero > 1) zero -= 1 - a[i++];
            ans = Math.max(ans, j - i);
        }
        return ans;
    }
}